### PR TITLE
SqlServer: resolve included properties of index to column names (in migrations)

### LINKED ISSUE #14087: ForSqlServerInclude does not respect column mapping
### Steps to reproduce
For this model and mapping:

```c#
	public class SomeTable
	{
		public int Id { get; set; }
		public int Value1 { get; set; }		
		public int Value2 { get; set; }
	}

	public class ApplicationDbContext : DbContext
	{
		protected override void OnModelCreating(ModelBuilder modelBuilder)
		{
			base.OnModelCreating(modelBuilder);

			modelBuilder.Entity<SomeTable>().ForSqlServerHasIndex(st => st.Value1).ForSqlServerInclude(st => st.Value2);

			modelBuilder.Entity<SomeTable>().Property(st => st.Id).HasColumnName("SomeTableId");
			modelBuilder.Entity<SomeTable>().Property(st => st.Value1).HasColumnName("SomeTableValue1");
			modelBuilder.Entity<SomeTable>().Property(st => st.Value2).HasColumnName("SomeTableValue2");
		}

		protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
		{
			base.OnConfiguring(optionsBuilder);

			optionsBuilder.UseSqlServer(@"Data Source=(localdb)\mssqllocaldb;Initial Catalog=EFCoreTest");
		}
	}
```
is created a migration
```c#
        protected override void Up(MigrationBuilder migrationBuilder)
        {
            migrationBuilder.CreateTable(
                name: "SomeTable",
                columns: table => new
                {
                    SomeTableId = table.Column<int>(nullable: false)
                        .Annotation("SqlServer:ValueGenerationStrategy", SqlServerValueGenerationStrategy.IdentityColumn),
                    SomeTableValue1 = table.Column<int>(nullable: false),
                    SomeTableValue2 = table.Column<int>(nullable: false)
                },
                constraints: table =>
                {
                    table.PrimaryKey("PK_SomeTable", x => x.SomeTableId);
                });

            migrationBuilder.CreateIndex(
                name: "IX_SomeTable_SomeTableValue1",
                table: "SomeTable",
                column: "SomeTableValue1")
                .Annotation("SqlServer:Include", new[] { "Value2" });
        }
```
See the difference: column: "**SomeTableValue1**" vs Annotation(..."**Value2**" }).

When scripted, a broken SQL Script is generated (column Value2 used in the index does not exists in the table).
```sql
CREATE TABLE [SomeTable] (
    [SomeTableId] int NOT NULL IDENTITY,
    [SomeTableValue1] int NOT NULL,
    [SomeTableValue2] int NOT NULL,
    CONSTRAINT [PK_SomeTable] PRIMARY KEY ([SomeTableId])
);

GO

CREATE INDEX [IX_SomeTable_SomeTableValue1] ON [SomeTable] ([SomeTableValue1]) INCLUDE ([Value2]);

GO
```

### Further technical details
EF Core version: 2.2.0
Database Provider: Microsoft.EntityFrameworkCore.SqlServer

### PR DESCRIPTION
**Pull request by mirol-h**
Fixes #14087

### CODE REVIEW

File: src/EFCore.SqlServer/Migrations/Internal/SqlServerMigrationsAnnotationProvider.cs [hunk 1]
```diff
@@ -88,9 +88,13 @@ public override IEnumerable<IAnnotation> For(IIndex index)
             var includeProperties = index.GetIncludeProperties();
             if (includeProperties != null)
             {
+                var includeColumns = (IReadOnlyList<string>)includeProperties
+                    .Select(p => index.DeclaringEntityType.FindDeclaredProperty(p).GetColumnName())
```
Comment (AndriySvyryd) [inline src/EFCore.SqlServer/Migrations/Internal/SqlServerMigrationsAnnotationProvider.cs]: Should be 'FindProperty' to handle inherited properties

File: test/EFCore.SqlServer.Tests/Migrations/SqlServerMigrationAnnotationProviderTest.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
+﻿using System.Linq;
+using Microsoft.EntityFrameworkCore.Infrastructure;
+using Microsoft.EntityFrameworkCore.SqlServer.Metadata.Internal;
+using Microsoft.EntityFrameworkCore.SqlServer.Migrations.Internal;
+using Microsoft.EntityFrameworkCore.TestUtilities;
+using Xunit;
+
+namespace Microsoft.EntityFrameworkCore.Migrations
+{
+    public class SqlServerMigrationAnnotationProviderTest
```
Comment (bricelam) [inline test/EFCore.SqlServer.Tests/Migrations/SqlServerMigrationAnnotationProviderTest.cs]: I recently added this too. Should be there if you sync changes from release/3.0

Comment (mirol-h) [reply test/EFCore.SqlServer.Tests/Migrations/SqlServerMigrationAnnotationProviderTest.cs]: Sorry about that. I synced release/3.0 in my fork.

Should I rebase/merge or leave it as is?

Comment (bricelam) [reply test/EFCore.SqlServer.Tests/Migrations/SqlServerMigrationAnnotationProviderTest.cs]: I too rebase and force-push; GitHub handles it just fine... 😎

Comment (mirol-h) [reply test/EFCore.SqlServer.Tests/Migrations/SqlServerMigrationAnnotationProviderTest.cs]: Rebased on most recent version from release/3.0.

**Pull request merged**